python/lsst/resources/http.py

```diff
@@ -602,6 +871,41 @@ def _openImpl(
                 yield http_handle
 
 
+def _dump_response(resp: requests.Response) -> None:
+    """Log the contents of a HTTP or webDAV request and its response.
+
+    Parameters
+    ----------
+    resp : `requests.Response`
+        The response to log.
+
+    Notes
+    -----
+    Intended for development purposes only.
+    """
+    log.debug("-----------------------------------------------")
+    log.debug("Request")
+    log.debug(f"   method={resp.request.method}")
+    log.debug(f"   URL={resp.request.url}")
+    log.debug(f"   headers={resp.request.headers}")
+    if resp.request.method == "PUT":
+        log.debug("   body=<data>")
+    elif resp.request.body is None:
+        log.debug("   body=<empty>")
+    else:
+        log.debug(f"   body={resp.request.body[:120]!r}")
```

Comment:
Do not use f-strings in `log.debug` calls. This should be:

```python
log.debug("   body=%r", resp.request.body[:120])
```

Reply:
Fixed.